src/libraries/System.Formats.Tar/src/System/Formats/Tar/PosixTarEntry.cs

```diff
@@ -10,16 +10,40 @@ namespace System.Formats.Tar
     /// Even though the <see cref="TarEntryFormat.Gnu"/> format is not POSIX compatible, it implements and supports the Unix-specific fields that were defined in that POSIX standard.</remarks>
     public abstract partial class PosixTarEntry : TarEntry
     {
-        // Constructor used when reading an existing archive.
-        internal PosixTarEntry(TarHeader header, TarReader readerOfOrigin)
-            : base(header, readerOfOrigin)
+        // Constructor called when reading a TarEntry from a TarReader.
+        internal PosixTarEntry(TarHeader header, TarReader readerOfOrigin, TarEntryFormat format)
+            : base(header, readerOfOrigin, format)
         {
         }
 
-        // Constructor called when creating a new 'TarEntry*' instance that can be passed to a TarWriter.
         internal PosixTarEntry(TarEntryType entryType, string entryName, TarEntryFormat format)
             : base(entryType, entryName, format)
         {
+            _header._uName = string.Empty;
+            _header._gName = string.Empty;
+            _header._devMajor = 0;
+            _header._devMinor = 0;
+        }
+
+        // Constructor called when creating a new TarEntry.
```

Comment:
Are you adding these comments to be useful, or just to match some sort of style where every method has a comment?

If there's an element of "everything should have a comment", I recommend repositioning to "add comments to non-public(/protected) members only when they add value" (or, the other extreme, xmldoc everything, I suppose).

Either way, this comment is either vacuous (constructors are of course called when creating a new TarEntry) or just has a misplaced verb ("Constructor called when converting an existing TarEntry" -- and this is pretty apparent from its signature, returning to "what's the value in this comment?")

Reply:
Just a poor comment. I'll improve it, and also add the missing one to the constructor above.